### PR TITLE
added subset argument in readSource

### PR DESCRIPTION
**Pull request by tscheypidi**

### CODE REVIEW

**Review by FelicitasBeier — APPROVED** [review]
Looks good

**Comment by FelicitasBeier** [comment]
The code check complains about codecov/patch. Not sure if this is a problem?

**Comment by k4rst3ns** [comment]
Would be happy to merge soon.
@tscheypidi: Let us now if you want to add additional tests or if we could merge with lower code coverage.

**Comment by tscheypidi** [comment]
lower code coverage is ok for the moment I think. I am just waiting for a review from @pfuehrlich-pik

File: R/readSource.R
Comment (pfuehrlich-pik) [inline R/readSource.R:14]:
```suggestion
#' data sets where reading in the whole data set might be impractical or even
```

**Review by pfuehrlich-pik — APPROVED** [review]

**Pull request merged**
